src/lib.rs

```diff
@@ -111,3 +112,42 @@ where
         Ord::cmp(self, key.borrow())
     }
 }
+
+/// `ComparableRangeBounds` is implemented as an extention to `RangeBounds` to
+/// allow for comparison of items with range bounds.
+pub trait ComparableRangeBounds<T: ?Sized>: RangeBounds<T> {
+    /// Returns `true` if `item` is contained in the range.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// assert!( (3..5).comparable_contains(&4));
+    /// assert!(!(3..5).comparable_contains(&2));
+    ///
+    /// assert!( (0.0..1.0).comparable_contains(&0.5));
+    /// assert!(!(0.0..1.0).comparable_contains(&f32::NAN));
+    /// assert!(!(0.0..f32::NAN).comparable_contains(&0.5));
+    /// assert!(!(f32::NAN..1.0).comparable_contains(&0.5));
```

Comment:
`f32` won't work because the blanket `Comparable` requires `Ord`.

Maybe a `str`/`String` example would be better to demonstrate the borrowing relationship?

Reply:
I move the example to the crate-level documentation.